File: python/day1.py
def readInput(filename):
    with open(filename) as file:
        lines = file.readlines()
        lines = [line.rstrip() for line in lines]
    return lines

def part1(lineList):
    elves = []
    elf = 0
    for i in lineList:
        if i != '':
            elf += int(i)
        else:
            elves.append(elf)
            elf = 0
    elves.sort()
    return elves[-1]


def part2(lineList):
    elves = []
    elf = 0
    for i in lineList:
        if i != '':
            elf += int(i)
        else:
            elves.append(elf)
            elf = 0
    elves.sort()
    return elves[-1] + elves[-2] + elves[-3]

if __name__ == '__main__':
    lineList = readInput('day1.txt')
    print(part1(lineList))
    print(part2(lineList))

File: c++/src/days/Day1.cpp
#include <iostream>
#include <string>
#include <algorithm>

#include "Day1.h"
#include "../helpers/Reader.h"

Day1::Day1()
{
	//m_vecInput = ReadInput("../../../inputs/day1.txt");
	m_vecInput = ReadInput("../inputs/day1.txt");

}

int Day1::Part1(std::vector<int> elves)
{

	
	std::vector<int>::iterator iterelves;
	int max = 0;
	for(iterelves = elves.begin(); iterelves < elves.end(); iterelves++ )
	{
		if(*iterelves > max)
		{
			max = *iterelves;
		}
	}
	return max;
}

int Day1::Part2(std::vector<int> elves)
{
	std::sort(elves.begin(),elves.end(),std::greater<int>());
	return elves[0] + elves[1] + elves[2];
}

bool Day1::Solve()
{
	std::vector<int> elves;

	std::vector<std::string>::iterator iter;
	int currelf = 0;
	for(iter = m_vecInput.begin(); iter < m_vecInput.end(); iter++ )
	{
		if(!iter->empty())
		{
			currelf += stoi(*iter);
		}
		else
		{
			elves.push_back(currelf);
			currelf = 0;
		}
	}
	elves.push_back(currelf);

	std::cout << Part1(elves) <<std::endl;
	std::cout << Part2(elves) <<std::endl;

	return true;
}